### PR TITLE
initial POC for grpc tls

### PR DESCRIPTION
**Pull request by jhawk28**
Initial concept for gRPC tls support

### CODE REVIEW

**Comment by a26nine** [comment]
I was able to sync a light node with:
```
[Core]
  IP = "xxx.celestia-mainnet.quiknode.pro"
  RPCPort = "443"
  GRPCPort = "9090"
  EnableTLS = true
```
using this build.

File: state/core_access.go [hunk 1]
```diff
@@ -124,9 +119,15 @@ func (ca *CoreAccessor) Start(ctx context.Context) error {
 
 	// dial given celestia-core endpoint
 	endpoint := fmt.Sprintf("%s:%s", ca.coreIP, ca.grpcPort)
+	grpcOpts := make([]grpc.DialOption, 0)
+	if ca.enableTLS {
+		grpcOpts = append(grpcOpts, grpc.WithTransportCredentials(credentials.NewTLS(&tls.Config{})))
```
Comment (vgonkivs) [inline state/core_access.go:124]: Without specifying valid certificates in the tls.Config object will not secure gRPC connection. By default, an empty tls.Config{} means that no certificate or private key is provided, which leads to an insecure setup.

Node already has TLS certificates handling logic [here](https://github.com/celestiaorg/celestia-node/blob/ace6840bb00b337c320c07be4dbe10e4d853271d/nodebuilder/p2p/tls.go#L24) for web socket, so you can reuse the certificate for the grpc connection.

File: nodebuilder/core/config.go [hunk 2]
```diff
@@ -16,9 +16,10 @@ var MetricsEnabled bool
 
 // Config combines all configuration fields for managing the relationship with a Core node.
 type Config struct {
-	IP       string
-	RPCPort  string
-	GRPCPort string
+	IP        string
+	RPCPort   string
+	GRPCPort  string
+	EnableTLS bool
```
Comment (vgonkivs) [inline nodebuilder/core/config.go:22]: let's avoid growing the node's config. We have an env variable that signals that TLS should be enabled.

**Review by vgonkivs — CHANGES_REQUESTED** [review]

**Comment by jcstein** [comment]
is it possible to set it up the same way that `rpc-mocha.pops.one` (mocha) works, for example? it uses the light node's default config and works by default (`celestia light start --p2p.network mocha --core.ip rpc-mocha.pops.one`)

File: state/core_access.go (hunk 1 above)
Comment (jhawk28) [reply state/core_access.go:124]: This was a proof of concept to get the Light node working against a Quicknode celestia endpoint. QN doesn't support client certs. gRPC auth for a QN endpoint occurs over the `x-token` header

Comment (vgonkivs) [reply state/core_access.go:124]: One more point here is the solution is not generic. You may not support the certificates but if someone does, the connection won't be established. Also, it is recommended to manually specify the MinTLSVersion(linter confirmed it)

**Comment by vgonkivs** [comment]
closed in favor of #3954

**Pull request closed without merging**